### PR TITLE
Direct deployment to other instance.

### PR DESCRIPTION
**Pull request by v-prkali**

### CODE REVIEW

File: CRM Solution Manager/CRM Solution/CrmSolutionHelper.cs [hunk 1]
```diff
@@ -319,6 +327,135 @@ private void ExportListOfSolutionsToBeMerged(OrganizationServiceProxy servicePro
...
+                    case 21:
+                        break;
+                    //Canceling
+                    case 22:
+                        break;
+                    //Failed
+                    case 31:
+                        break;
+                    //Canceled
+                    case 32:
```
Comment (amrullah08) [inline CRM Solution Manager/CRM Solution/CrmSolutionHelper.cs]: for case 21,22,31 log in console log, abort the entire execution if it fails here.

File: CRM Solution Manager/CRM Solution/CrmSolutionHelper.cs [hunk 2]
```diff
@@ -319,6 +327,135 @@ private void ExportListOfSolutionsToBeMerged(OrganizationServiceProxy servicePro
             }
         }
 
+        /// <summary>
+        /// Method gets deployment instance record
+        /// </summary>
+        /// <param name="serviceProxy">organization service proxy</param>
+        /// <param name="solutionFile">solution file info</param>        
+        private void GetDeploymentInstance(OrganizationServiceProxy serviceProxy, SolutionFileInfo solutionFile)
```
Comment (amrullah08) [inline CRM Solution Manager/CRM Solution/CrmSolutionHelper.cs]: rename method to ImportSolutionToTargetInstance

File: CRM Solution Manager/CRM Solution/CrmSolutionHelper.cs [hunk 3]
```diff
@@ -319,6 +327,135 @@ private void ExportListOfSolutionsToBeMerged(OrganizationServiceProxy servicePro
...
+
+        /// <summary>
+        /// Method import master solution to deployment instance
+        /// </summary>
+        /// <param name="serviceProxy">organization service proxy</param>
+        /// <param name="solutionImportPath">solution import path</param>
+        public static void ImportSolution(OrganizationServiceProxy serviceProxy, string solutionImportPath)
+        {
+            byte[] fileBytes = File.ReadAllBytes(solutionImportPath);
+
```
Comment (amrullah08) [inline CRM Solution Manager/CRM Solution/CrmSolutionHelper.cs]: you need to write in the log, starting deployment ot the organization + organizaitonserviceurl, same way completed deployment to the instance as well

File: CRM Solution Manager/CRM Solution/CrmSolutionHelper.cs [hunk 4]
```diff
@@ -319,6 +327,135 @@ private void ExportListOfSolutionsToBeMerged(OrganizationServiceProxy servicePro
...
+            };
+            ExecuteAsyncResponse importRequestResponse = (ExecuteAsyncResponse)serviceProxy.Execute(importRequest);
+
+            Console.WriteLine("Started importing solution");
+
+            // put in sleep for every 30 seconds to check the status of import using asyncoperation entity
+            string solutionImportResult = null;
+            while (solutionImportResult == null)
+            {
+                System.Threading.Thread.Sleep(30000);
```
Comment (amrullah08) [inline CRM Solution Manager/CRM Solution/CrmSolutionHelper.cs]: remove this wait
make it as synchronous operation

File: CRM Solution Manager/CRM Solution/CrmSolutionHelper.cs [hunk 5]
```diff
@@ -319,6 +327,135 @@ private void ExportListOfSolutionsToBeMerged(OrganizationServiceProxy servicePro
...
+        /// <param name="serviceProxy">organization service proxy</param>
+        /// <param name="solutionFile">solution file info</param>        
+        private void GetDeploymentInstance(OrganizationServiceProxy serviceProxy, SolutionFileInfo solutionFile)
+        {
+            Entity sourceControl = solutionFile.Solution;
+            EntityCollection deploymentInstance = FetchDeplopymentInstance(serviceProxy, sourceControl.Id);
+
+            if (deploymentInstance.Entities.Count > 0)
+            {
+                foreach (Entity instance in deploymentInstance.Entities)
```
Comment (amrullah08) [inline CRM Solution Manager/CRM Solution/CrmSolutionHelper.cs]: you can do threading here, use parallel.foreach

Comment (v-prkali) [reply CRM Solution Manager/CRM Solution/CrmSolutionHelper.cs]: In future. we will start using parallel.foreach.

**Pull request merged**